ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/ScanConfigs/Wizard/useFormikScanConfig.tsx

```diff
@@ -33,7 +33,13 @@ export const defaultScanConfigFormValues: ScanConfigFormValues = {
 
 const validationSchema = yup.object().shape({
     parameters: yup.object().shape({
-        name: yup.string().required('Scan name is required'),
+        name: yup
+            .string()
+            .required('Scan name is required')
+            .matches(
+                /(^((?!-)(?!\.)[a-z0-9-.]{0,253}$))/,
+                "Name can contain only lowercase alphanumeric characters, '-' or '.', and start and end with an alphanumeric character."
```

Comment:
Does this satisfy the constraint "end with an alphanumeric character"? From what I can tell that isn't the case:
```ts
/(^((?!-)(?!\.)[a-z0-9-.]{0,253}$))/.test("scan.");
// true
```

Reply:
Doh! I was so concerned with the negative lookahead at the front, I forgot about the back. 

In a effort not to complicate it any more, I think I will instead go with something simpler, like this
`^[a-z0-9][a-z0-9.-]{0,251}[a-z0-9]$`

Address the ending character limitation, although it does require at least 2 characters for the name.  (probably a good practice anyway)

<img width="409" alt="Screenshot 2024-01-19 at 4 53 06 PM" src="https://github.com/stackrox/stackrox/assets/715729/ea03b5c4-efef-4192-b7cf-d1240a99edbd">